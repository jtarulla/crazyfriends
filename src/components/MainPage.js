import React, { Component } from 'react';

import CardList from './CardList';
import SearchBox from './SearchBox';
import ErrorBoundry from './ErrorBoundry';
import CounterButton from './CounterButton';
import './MainPage.css';


class MainPage extends Component {
  
  componentDidMount() {
    this.props.onRequestFriends()
  } 
  
  filteredFriends = () => {
    return this.props.friends.filter(friend => {
      return friend.name.toLowerCase().includes(this.props.searchField.toLowerCase())
  })}

  render() {
    const { onSearchChange, isPending } = this.props;
   
    return isPending ?
    <h1 id='loading-page' className='tc f1'>loading</h1> :
    (
      <div className='tc'>
        <h1 className='f1'> crazyfriends </h1>
        <CounterButton />
        <SearchBox searchChange={onSearchChange}/>
          <ErrorBoundry>
            <CardList friends={this.filteredFriends()}/>
          </ErrorBoundry>
      </div>
    )
  }
}

export default MainPage
