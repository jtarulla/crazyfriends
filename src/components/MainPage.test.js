import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

const mockProps = {
  onRequestFriends: jest.fn(),
  friends: [{
    id: 1, 
    name: 'Kristina Doe',
    username: 'Kristi',
    email: 'kristinadoe@email.com'
  },
  {
    id: 2, 
    name: 'John Doe',
    username: 'John-O',
    email: 'johndoe@email.com'
  }],
  searchField: 'K',
  isPending: true
}

let wrapper = shallow(<MainPage {...mockProps}/>)

it('renders MainPage component', () => {
  expect(wrapper).toMatchSnapshot();
})

it('filters friends properly', () => {
  expect(wrapper.instance().filteredFriends()).toEqual([{
    id: 1, 
    name: 'Kristina Doe',
    username: 'Kristi',
    email: 'kristinadoe@email.com'
  }])
})

