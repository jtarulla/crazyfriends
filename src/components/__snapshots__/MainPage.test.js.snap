// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage component 1`] = `ShallowWrapper {}`;
